src/datasets/domain/models/CitationResponse.ts

```diff
@@ -0,0 +1,4 @@
+export type CitationResponse = {
+  content: string | object
```

Comment:
Since we are doing a client-side download,  I think we will have to stringify the json object if type = application/json.  So rather than return either a string or object, it might be more straightforward to stringify the json object response from the API, and content then can always be type string.
```suggestion
  content: string
```